MSAL/test/integration/native_auth/end_to_end/mfa/MSALNativeAuthSignInWithMFAEndToEndTests.swift

```diff
@@ -0,0 +1,226 @@
+        mfaRequiredState.requestChallenge(authMethod: authMethod, delegate: mfaSendChallengeDelegateSpy)
+        
+        await fulfillment(of: [mfaSendChallengeExpectation])
+        
+        guard mfaSendChallengeDelegateSpy.onVerificationRequiredCalled, let mfaRequiredState = mfaSendChallengeDelegateSpy.newStateMFARequired else {
+            XCTFail("Challenge not sent to MFA method")
+            return
+        }
+        
+        // Now retrieve and submit the email OTP code
+        await retrieveAndSubmitCode(state: mfaRequiredState, username: username)
+    }
+    
+    func test_signInUsingPasswordWithMFANoDefaultAuthMethod_completeSuccessfully() async throws {
+        throw XCTSkip("Missing username for MFA user with no default auth method")
+        guard let username = retrieveUsernameForSignInUsernamePasswordAndMFANoDefaultAuthMethod(),
+                let password = await retrievePasswordForSignInUsername(),
+                let awaitingMFAState = await signInUsernameAndPassword(username: username, password: password)
+        else {
+            XCTFail("Something went wrong")
+            return
+        }
+        
+        // Request to send challenge to the default strong auth method
+        let mfaExpectation = expectation(description: "mfa")
+        let mfaDelegateSpy = MFARequestChallengeDelegateSpy(expectation: mfaExpectation)
+        
+        awaitingMFAState.requestChallenge(delegate: mfaDelegateSpy)
+        
+        await fulfillment(of: [mfaExpectation])
+        
+        guard mfaDelegateSpy.onSelectionRequiredCalled, let mfaRequiredState = mfaDelegateSpy.newStateMFARequired, let authMethod = mfaDelegateSpy.authMethods?.first else {
+            XCTFail("Selection required not triggered")
+            return
+        }
+        
+        XCTAssertTrue(authMethod.channelTargetType.isEmailType)
+        
+        // Request to send challenge to a specific strong auth method
+        
+        let mfaSendChallengeExpectation = expectation(description: "mfa")
+        let mfaSendChallengeDelegateSpy = MFARequestChallengeDelegateSpy(expectation: mfaSendChallengeExpectation)
+        mfaRequiredState.requestChallenge(authMethod: authMethod, delegate: mfaSendChallengeDelegateSpy)
+        
+        await fulfillment(of: [mfaSendChallengeExpectation])
+        
+        guard mfaSendChallengeDelegateSpy.onVerificationRequiredCalled, let mfaRequiredState = mfaSendChallengeDelegateSpy.newStateMFARequired else {
+            XCTFail("Challenge not sent to MFA method")
+            return
+        }
+        
+        // Now retrieve and submit the email OTP code
+        await retrieveAndSubmitCode(state: mfaRequiredState, username: username)
+    }
+    
+    // MARK: private methods
+    
+    private func signInUsernameAndPassword(username: String, password: String) async -> AwaitingMFAState? {
+        guard let sut = initialisePublicClientApplication()
```

Comment:
`sut`?

Reply:
with "sut" we mean the system under test